src/Core/Settings.h

```diff
@@ -1145,6 +1145,7 @@ class IColumn;
     M(Bool, input_format_values_accurate_types_of_literals, true, "For Values format: when parsing and interpreting expressions using template, check actual type of literal to avoid possible overflow and precision issues.", 0) \
     M(Bool, input_format_values_allow_data_after_semicolon, false, "For Values format: allow extra data after semicolon (used by client to interpret comments).", 0) \
     M(Bool, input_format_avro_allow_missing_fields, false, "For Avro/AvroConfluent format: when field is not found in schema use default value instead of error", 0) \
+    M(Bool, cast_keys_to_string_from_json, false, "Either to cast string to int in enumerating json or not", 0) \
```

Comment:
As we will make it work only in non-ambiguous cases, we can set it to true by default.